File: HackerRank 30 Days/day19/main.py
class AdvancedArithmetic:
    def divisorSum(self, n):
        pass


class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        total = 0

        for i in range(1, n + 1):
            if n % i == 0:
                total += i

        return total


n = int(input())

myCalculator = Calculator()

print("I implemented: AdvancedArithmetic")
print(myCalculator.divisorSum(n))

File: HackerRank 30 Days/day19/main.cs
using System;

interface AdvancedArithmetic
{
    int divisorSum(int n);
}

class Calculator : AdvancedArithmetic
{
    public int divisorSum(int n)
    {
        int total = 0;

        for (int i = 1; i <= n; i++)
        {
            if (n % i == 0)
            {
                total += i;
            }
        }

        return total;
    }
}

class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());

        Calculator myCalculator = new Calculator();

        Console.WriteLine("I implemented: AdvancedArithmetic");
        Console.WriteLine(myCalculator.divisorSum(n));
    }
}